cdragontoolbox/binfile.py

```diff
@@ -294,6 +294,44 @@ def __repr__(self):
     def to_serializable(self):
         return (self.name.to_serializable(), {_to_serializable(k): _to_serializable(v) for k,v in self.value.items()})
 
+class BinNested(BinObjectWithFields):
+    """Nested binary value"""
+
+    def __init__(self, fields):
+        super().__init__(0, fields)
```

Comment:
To avoid this useless/hacky `0` you should define a new base class but without `htype`.
Typically:
- remove `type` from `BinObjectWithFields` and adjust its `to_serializable()` accordingly
- define a new `BinObjectWithFieldsAndType` derived from `BinObjectWithFields` and adding the `type` attribute
- change previous uses of `BinObjectWithFields` to now inherit from `BinObjectWithFieldsAndType`
- remove custom constructor from `BinNested` and `BinNestedField` (not needed anymore)